### PR TITLE
Add catalog support for Thingsboard 

### PR DESCRIPTION
**Pull request by danielfree**
[Thingsboard](https://thingsboard.io/) is an open-source IoT platform for data collection, processing, visualization, and device management. This PR adds support for thingsboard catalog with external zookeeper service link, two database backends choices ( postgres or cassandra) using conditional template, persistent volumes.

### CODE REVIEW

File: templates/thingsboard/0/docker-compose.yml.tpl [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      {{- if eq .Values.database_type "cassandra" }}
+      - cassandra
+      {{- else }}
+      - postgres
+      {{- end }} 
+    external_links:
+      - ${zookeeper_service}:zk    
+    entrypoint: /run-application.sh
+  {{- if eq .Values.database_type "cassandra" }}  
+  cassandra:
```
Comment (rawmind0) [inline templates/thingsboard/0/docker-compose.yml.tpl]: What to you thing to use a generic db service and just change image and volume??
```
-  {{- if eq .Values.database_type "cassandra" }}  
-  cassandra:  
+  db:
+  {{- if eq .Values.database_type "cassandra" }}
```

File: templates/thingsboard/0/docker-compose.yml.tpl [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    image: "cassandra:3"
+    volumes:
+      - cassandra_data_dir:/var/lib/cassandra
+  {{- else }}   
+  postgres:
```
Comment (rawmind0) [inline templates/thingsboard/0/docker-compose.yml.tpl]: Using generic db service
```
-  postgres:
```

File: templates/thingsboard/0/docker-compose.yml.tpl [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+volumes:
+  hsqldb_data_dir:
+    driver: ${volume_driver}
+  {{- if eq .Values.database_type "cassandra" }}  
+  cassandra_data_dir:
+    driver: ${volume_driver}
+  {{- else }}  
+  postgres_data_dir:
+    driver: ${volume_driver}
+  {{- end }}
```
Comment (rawmind0) [inline templates/thingsboard/0/docker-compose.yml.tpl]: Setting driver outside if section
```
  {{- if eq .Values.database_type "cassandra" }}  
  cassandra_data_dir:
-    driver: ${volume_driver}
  {{- else }}  
  postgres_data_dir:
-    driver: ${volume_driver}
  {{- end }}
+    driver: ${volume_driver}
```

File: templates/thingsboard/0/docker-compose.yml.tpl [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      - MQTT_BIND_PORT=1883
+      - COAP_BIND_ADDRESS=0.0.0.0
+      - COAP_BIND_PORT=5683
+      - ZOOKEEPER_URL=zk:2181
+      - DATABASE_TYPE=${database_type}
+      - CASSANDRA_URL=cassandra:9042
+      - CASSANDRA_HOST=cassandra
+      - CASSANDRA_PORT=9042
+      - POSTGRES_HOST=postgres
+      - POSTGRES_PORT=5432
```
Comment (rawmind0) [inline templates/thingsboard/0/docker-compose.yml.tpl]: What to you thing to use a generic db service and just change image and volume??
```
-      - CASSANDRA_URL=cassandra:9042
+      - CASSANDRA_URL=db:9042
-      - CASSANDRA_HOST=cassandra
+      - CASSANDRA_HOST=db
      - CASSANDRA_PORT=9042
-      - POSTGRES_HOST=postgres
+      - POSTGRES_HOST=db
      - POSTGRES_PORT=5432
```

File: templates/thingsboard/0/docker-compose.yml.tpl [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+      - ADD_SCHEMA_AND_SYSTEM_DATA=${add_schema_and_system_data}
+      - ADD_DEMO_DATA=${add_demo_data}
+    volumes:
+      - hsqldb_data_dir:/usr/share/thingsboard/data/sql
+    depends_on:
+      {{- if eq .Values.database_type "cassandra" }}
+      - cassandra
+      {{- else }}
+      - postgres
+      {{- end }}
```
Comment (rawmind0) [inline templates/thingsboard/0/docker-compose.yml.tpl]: Adding link to db. Just to manage dependencies and show the connection
```
+    links:
+      - db:db  
````

**Comment by rawmind0** [comment]
Hi @danielfree .. thanks so much for the PR....

I've tried the package. Deploying with cassandra is working ok but  deploying with postgres, getting this error logs...
```
8/9/2017 9:12:22 PMexport ZOOKEEPER_ZK_1_ENV_SERVICE_HOME='/opt/zk'
8/9/2017 9:12:22 PMexport _='/usr/bin/printenv'
8/9/2017 9:12:22 PMStarting 'Thingsboard' service...
8/9/2017 9:12:22 PMStarted [106]
8/9/2017 9:12:32 PM	at java.lang.reflect.Method.invoke(Method.java:498)
8/9/2017 9:12:32 PM	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48)
8/9/2017 9:12:32 PM	at org.springframework.boot.loader.Launcher.launch(Launcher.java:87)
8/9/2017 9:12:32 PM	at org.springframework.boot.loader.Launcher.launch(Launcher.java:50)
8/9/2017 9:12:32 PM	at org.springframework.boot.loader.PropertiesLauncher.main(PropertiesLauncher.java:521)
8/9/2017 9:12:32 PMCaused by: org.springframework.context.ApplicationContextException: Unable to start EmbeddedWebApplicationContext due to missing EmbeddedServletContainerFactory bean.
8/9/2017 9:12:32 PM	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.getEmbeddedServletContainerFactory(EmbeddedWebApplicationContext.java:189)
8/9/2017 9:12:32 PM	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.createEmbeddedServletContainer(EmbeddedWebApplicationContext.java:162)
8/9/2017 9:12:32 PM	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.onRefresh(EmbeddedWebApplicationContext.java:134)
8/9/2017 9:12:32 PM	... 16 common frames omitted
```
Could you please take a look??

**Comment by danielfree** [comment]
@rawmind0 thanks for your review, I'll try to test with postgres and change the compose file. Btw, does rancher-compose cli support .tpl file? Right now I have to push my commits to github and test it with my environment -  can't do it locally with rancher-compose since I've used tpl.

**Comment by danielfree** [comment]
@rawmind0  bug fixed, please take a look, thanks!

**Comment by rawmind0** [comment]
Thanks @danielfree. Now, the package is working well with postgres and cassandra... :)

Just one thing else. Could you please add a link from tb to db?? It'd force to start db before tb, ```depends_on``` doesn't  do it right...
image

Adding link.....
image

Also, graph view is more helpful....
image
 
Adding link....
image

**Comment by danielfree** [comment]
@rawmind0 appreciated it! link is added 👍

**Comment by rawmind0** [comment]
Thanks @danielfree ...

LGTM...

**Pull request merged**